crates/ty/src/main.rs

```diff
@@ -401,6 +401,12 @@ fn set_colored_override(color: Option<TerminalColor>) {
 fn setup_rayon() {
     ThreadPoolBuilder::default()
         .num_threads(max_parallelism().get())
+        // Use a reasonably large stack size to avoid running into stack overflows too easily. The
+        // size was chosen in such a way as to still be able to handle large expressions involving
+        // binary operators (x + x + … + x) both during the AST walk in semantic index building as
+        // well as during type checking. Using this stack size, we can handle handle expressions
+        // that are several times larger than the corresponding limits in existing type checkers.
+        .stack_size(16 * 1024 * 1024)
```

Comment:
It's probably worth noting that this does not generally increase memory usage by 16 MiB × num_threads. It's just the max. available stack space that a single thread can use.

Reply:
And 16MiB/thread is not even detectable, relative to the memory we need for source texts and AST, for any project large enough to care about memory usage.